### ISSUE TITLE
tvOS build fail saying lipo errors, but iOS and macOS pass, on Xcode 12

### ISSUE THREAD

**Issue by liuxuan30** [Carthage/Carthage#3120]
* carthage install method: [ ] .pkg, [X] homebrew, [ ] source
* `which carthage`:homebrew
* `carthage version`: 0.36.1
* `xcodebuild -version`: Xcode12.0.1 & Xcode 12.4
* Are you using `--no-build`? no
* Are you using `--no-use-binaries`? yes, via carthage.sh
* Are you using `--use-submodules`? no
* Are you using `--cache-builds`? no
* Are you using `--new-resolver`? no

Hi there, 

We have been using the cathrage.sh since Xcode 12 and it seems working great, but suddenly the tvOS failed starting in this month, but we changed nothing about the process.

**And furthermore, iOS and CI passed at the same time, with same code and setup**. It's really confusing for us how to fix it.

The full travis build record:https://travis-ci.org/github/danielgindi/Charts/builds/756550941

The error for tvOS build:
```
Processing Info.plist
▸ Running script 'Copy Carthage Frameworks'
❌  error: /Applications/Xcode-12.0.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: one of -create, -thin <arch_type>, -extract <arch_type>, -remove <arch_type>, -replace <arch_type> <file_name>, -verify_arch <arch_type> ... , -archs, -info, or -detailed_info must be specified
Testing failed:
	/Applications/Xcode-12.0.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: one of -create, -thin <arch_type>, -extract <arch_type>, -remove <arch_type>, -replace <arch_type> <file_name>, -verify_arch <arch_type> ... , -archs, -info, or -detailed_info must be specified
	Testing cancelled because the build failed.
** TEST FAILED **
The following build commands failed:
	PhaseScriptExecution Copy\ Carthage\ Frameworks /Users/travis/Library/Developer/Xcode/DerivedData/ChartsDemo-iOS-aeselevbqicktffmzvanvfzicpzm/Build/Intermediates.noindex/Charts.build/Debug-appletvsimulator/ChartsTests.build/Script-6BA68666BDA3FCF79C2A6801.sh
(1 failure)
rake aborted!
```

iOS passed: https://travis-ci.org/github/danielgindi/Charts/jobs/756550942
tvOS constantly fail: https://travis-ci.org/github/danielgindi/Charts/jobs/756550943

Our carthage script: https://github.com/danielgindi/Charts/blob/master/carthage.sh

the script we use: https://github.com/danielgindi/Charts/blob/master/.travis.yml

We have a script to run copy frameworks:https://github.com/danielgindi/Charts/blob/master/scripts/copy-carthage-frameworks.sh

Could anyone shed some lights?

**Comment by liuxuan30** [comment]
turning off xcpretty gives below:
```
/bin/sh -c /Users/xl/Library/Developer/Xcode/DerivedData/ChartsDemo-iOS-aumclvnnjijknmfjqpwftsmkrchx/Build/Intermediates.noindex/Charts.build/Debug-appletvsimulator/ChartsTests.build/Script-6BA68666BDA3FCF79C2A6801.sh
A shell task (/usr/bin/xcrun lipo /Users/xl/Development/GitHub/ios-charts/Carthage/Build/tvOS/FBSnapshotTestCase.framework//FBSnapshotTestCase -output /var/folders/8v/6dg7d6mx2850sv1gp8ts9thm0000gn/T/carthage-lipo-VZRwrb/FBSnapshotTestCase) failed with exit code 1:
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: one of -create, -thin <arch_type>, -extract <arch_type>, -remove <arch_type>, -replace <arch_type> <file_name>, -verify_arch <arch_type> ... , -archs, -info, or -detailed_info must be specified
usage: lipo <input_file> <command> [<options> ...]
  command is one of:
    -archs
    -create
    -detailed_info
    -extract <arch_type> [-extract <arch_type> ...]
    -extract_family <arch_type> [-extract_family <arch_type> ...]
    -info
    -remove <arch_type> [-remove <arch_type> ...]
    -replace <arch_type> <file_name> [-replace <arch_type> <file_name> ...]
    -thin <arch_type>
    -verify_arch <arch_type> ...
  options are one or more of:
    -arch <arch_type> <input_file>
    -hideARM64
    -output <output_file>
    -segalign <arch_type> <alignment>



Test session results, code coverage, and logs:
	/Users/xl/Library/Developer/Xcode/DerivedData/ChartsDemo-iOS-aumclvnnjijknmfjqpwftsmkrchx/Logs/Test/Run-ChartsTests-2021.02.01_09-34-11-+0800.xcresult

Testing failed:
	/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: one of -create, -thin <arch_type>, -extract <arch_type>, -remove <arch_type>, -replace <arch_type> <file_name>, -verify_arch <arch_type> ... , -archs, -info, or -detailed_info must be specified
	Testing cancelled because the build failed.

** TEST FAILED **
```
it seems this error comes from carthage cannot run the command correctly for tvOS, but iOS is totally fine. Don't know why.

**Comment by daveklotz** [comment]
Seeing the exact same thing, though it did not start until we moved to 0.37.0.

**Comment by danielrbrowne** [comment]
I've seen this issue too - I flagged this on the (seemingly) related issue: https://github.com/Carthage/Carthage/issues/3112

**Comment by liuxuan30** [comment]
@danielrbrowne any idea why iOS is fine, tvOS suffers?

**Comment by elliottwilliams** [comment]
I think this should be fixed as of #3155 and v0.38.0. Feel free to reopen if you're still having issues.
